addons/account_check_printing/models/account_payment.py

```diff
@@ -85,14 +85,14 @@ def _compute_check_amount_in_words(self):
             else:
                 pay.check_amount_in_words = False
 
-    @api.depends('journal_id', 'payment_method_code')
+    @api.depends('journal_id', 'payment_method_code', 'state')
     def _compute_check_number(self):
-        for pay in self:
-            if pay.journal_id.check_manual_sequencing and pay.payment_method_code == 'check_printing':
+        for pay in self.filtered(lambda p: p.state == 'posted'):
+            if pay.journal_id.check_manual_sequencing and pay.payment_method_code == 'check_printing' and not pay.check_number:
                 sequence = pay.journal_id.check_sequence_id
-                pay.check_number = sequence.get_next_char(sequence.number_next_actual)
+                pay.check_number = sequence.next_by_id()
```

Comment:
If you are changing it in here, you should do the same in `account.journal`
https://github.com/odoo/odoo/blob/01cc43b0578ecc9d1fed37a12b2468ffc9d4aedd/addons/account_check_printing/models/account_journal.py#L39